scripts/pushArtifact.js

```diff
@@ -0,0 +1,145 @@
+    const result = { stdout: [], stderr: [] }
+    const cmdOptions = { encoding: 'utf8', ...options }
+    const process = await spawn(cmd, params, cmdOptions)
+    process.stdout.on('data', data => result.stdout.push(data.toString()))
+    process.stderr.on('data', data => result.stderr.push(data.toString()))
+    process.on('close', code => {
+      result.code = code
+      if (code === 0) {
+        resolve(result.stdout.join(''))
+      } else {
+        reject(new Error(result.stderr.join('')))
+      }
+    })
+  })
+}
+const updateVersion = async folderName => {
+  const buildCommit = process.env.TRAVIS_COMMIT
+  const TRAVIS_BUILD_DIR = process.env.TRAVIS_BUILD_DIR
+  const TRAVIS_BUILD_ID = process.env.TRAVIS_BUILD_ID
+
+  const pathToManifest = path.join(
+    TRAVIS_BUILD_DIR,
+    folderName,
+    'manifest.webapp'
+  )
+  const appManifestObj = fs.readJSONSync(pathToManifest)
+  const appVersion = `${
+    appManifestObj.version
+  }-dev.${buildCommit}-${TRAVIS_BUILD_ID}`
+  appManifestObj.version = appVersion
+  console.warn(`↳ ℹ️  Updating manifest version to ${appVersion}`)
+
+  fs.writeFileSync(pathToManifest, JSON.stringify(appManifestObj))
+}
+
+const createArchive = async (folderName, archiveFileName) => {
+  console.warn(`↳ ℹ️  Creating archive ${folderName}/${archiveFileName}`)
+  await updateVersion(folderName)
+  const fileList = await fs.readdir(folderName)
+  const options = {
+    gzip: true,
+    cwd: folderName,
+    file: folderName + '/' + archiveFileName
+  }
+  try {
+    await tar.c(options, fileList)
+  } catch (error) {
+    console.error(
+      `↳ ❌ Unable to generate app archive. Is tar installed as a dependency ? Error : ${
+        error.message
+      }`
+    )
+    throw new Error('Unable to generate archive')
+  }
+}
+
+const pushArtifact = async (fileName, parentFolder, options) => {
+  const { appSlug, appVersion, buildCommit } = options
+  console.warn(`↳ ℹ️  Sending artifact to downcloud`)
+  const folder = `${appSlug}/mobile/${appVersion}${
+    buildCommit ? `-${buildCommit}` : ''
+  }/`
+  try {
+    await launchCmd('ssh', [
+      '-o',
+      'StrictHostKeyChecking=no',
+      'upload@downcloud.cozycloud.cc',
+      `mkdir -p ${DOWNCLOUD_UPLOAD_DIR}${folder}`
+    ])
+  } catch (e) {
+    throw new Error(
+      `Unable to create target directory ${folder} on downcloud server : ${
+        e.message
+      }`
+    )
+  }
+
+  await launchCmd(
+    'rsync',
+    [
+      // to remove host validation question on CI
+      '-e',
+      'ssh -o StrictHostKeyChecking=no',
+      '-a',
+      fileName,
+      `upload@${DOWNCLOUD_URL}:${DOWNCLOUD_UPLOAD_DIR}${folder}`
+    ],
+    { cwd: parentFolder }
+  )
+
+  console.warn(`↳ ℹ️  Upload to downcloud complete.`)
+  return {
+    ...options,
+    appBuildUrl: `https://${DOWNCLOUD_URL}/upload/${folder}${fileName}`
+  }
+}
+
+const getUploadTarget = async () => {
+  const uploadTarget = process.argv[2]
```

Comment:
Access to command line arguments process.argv[2] should be done earlier, not in this function.

```
   const uploadTarget = process.argv[2]
   getUploadTarget(uploadTarget)
```